tests/e2e/specs/settings/applications-passwords.test.js

```diff
@@ -0,0 +1,103 @@
+import {
+    visitAdminPage,
+    __experimentalRest as rest,
+} from '@wordpress/e2e-test-utils';
+
+async function getResponseForApplicationPassword() {
+    return await rest({
+        method: 'GET',
+        path: '/wp/v2/users/me/application-passwords'
+    });
+}
+
+async function createApplicationPassword(applicationName) {
+    await visitAdminPage('profile.php');
+    await page.waitForSelector('#new_application_password_name');
+    await page.type('#new_application_password_name', `${applicationName}`);
+    await page.click('#do_new_application_password');
+
+    await page.waitForSelector('.notice');
+}
+
+async function revokeAllApplicationPasswords() {
+    await visitAdminPage('profile.php');
+
+    const revokeAllButtonVisibility = await page.evaluate(() => {
+        const e = document.querySelector('.application-passwords-list-table-wrapper');
+        if (!e) {
+            return false;
+        }
+        const visibility = window.getComputedStyle(e);
+        return visibility && visibility.display !== 'none' && visibility.visibility !== 'hidden' && visibility.opacity !== '0';
+    });
+
+    if (revokeAllButtonVisibility) {
+        await page.click('#revoke-all-application-passwords');
+        await page.keyboard.press('Enter');
+        await page.waitForSelector('.notice-success');
+    }
+}
+
+describe('Manage applications passwords', () => {
+    const testApplicationName = 'Test Application';
+
+    beforeEach(async() => {
+        await revokeAllApplicationPasswords();
+    });
+
+    it('correctly creates a new application password', async() => {
+        await createApplicationPassword(testApplicationName);
+
+        const response = await getResponseForApplicationPassword();
+        expect(response[0]['name']).toBe(testApplicationName);
+
+        const successMessage = await page.waitForSelector('.notice-success');
+        expect(
+            await successMessage.evaluate((element) => element.textContent)
+        ).toContain('Your new password for Test Application is:');
+
+        expect(
+            await successMessage.evaluate((element) => element.textContent)
+        ).toContain('Be sure to save this in a safe location. You will not be able to retrieve it.');
```

Comment:
We can use [`toMatch`](https://jestjs.io/docs/27.1/expect#tomatchregexp--string) here to assert for RegExp and combine these two assertions. A bonus benefit would be to be able to assert for the pattern of the application password, so that it won't be `undefined` or `NaN` for instance.